base/math.jl

```diff
@@ -72,6 +77,13 @@ function clamp!(x::AbstractArray, lo, hi)
     x
 end
 
+function clamp!(x::AbstractArray, ::Type{T}) where {T}
```

Comment:
not strictly related, but is the `clamp!` function even necessary? Seems to me it could be written

```
A .= clamp.(lo, A, hi)
```

instead of

```
clamp!(A, lo, hi)
```